server/src/main/java/io/unitycatalog/server/persist/dao/CommitDAO.java

```diff
@@ -0,0 +1,83 @@
+package io.unitycatalog.server.persist.dao;
+
+import io.unitycatalog.server.model.Commit;
+import io.unitycatalog.server.model.CommitInfo;
+import jakarta.persistence.*;
+import java.util.Date;
+import java.util.UUID;
+import lombok.*;
+import org.hibernate.annotations.UuidGenerator;
+
+@Entity
+@Table(
+    name = "uc_commits",
+    uniqueConstraints = {
+      @UniqueConstraint(columnNames = {"metastore_id", "table_id", "commit_version"})
+    })
+// Lombok
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class CommitDAO {
+  @Id
+  @UuidGenerator
+  @Column(name = "id", columnDefinition = "BINARY(16)")
+  private UUID id;
+
+  @Column(name = "table_id", nullable = false, columnDefinition = "BINARY(16)")
+  private UUID tableId;
+
+  @Column(name = "commit_version", nullable = false)
+  private Long commitVersion;
+
+  @Column(name = "commit_filename", nullable = false)
+  private String commitFilename;
+
+  @Column(name = "commit_filesize", nullable = false)
+  private Long commitFilesize;
+
+  @Column(name = "commit_file_modification_timestamp", nullable = false)
+  private Date commitFileModificationTimestamp;
+
+  @Column(name = "commit_timestamp", nullable = false)
+  private Date commitTimestamp;
+
+  @Column(name = "is_backfilled_latest_commit")
```

Comment:
Why is it nullable?

Reply:
For non latest backfilled commits, this field is null right now. We could make `nullable = false` and enforce that this field is always populated with false by default if that would be better.

Reply:
Yeah, let's do that.